### PR TITLE
Bug fixes (Issues #151 and #302 from original repo)

### PR DESCRIPTION
**Pull request by LissaGreense**
Hi,

I fixed some bugs :smile: Most of them have been reported in original repository (below u can find links).

- https://github.com/etingof/pysnmp/issues/302
- https://github.com/etingof/pysnmp/issues/151 (this one was fixed by [etingof](https://github.com/etingof), but code has never been merged)
- Hex printing in debug mode ( I noticed this one during tests using Python 3.9.10. I didn't investigate it a lot )

### CODE REVIEW

**Comment by LissaGreense** [comment]
Could I ask for review? :smile:

**Comment by ryanfaircloth** [comment]
@omrozowicz-splunk got time to review this one?

**Comment by omrozowicz-splunk** [comment]
> Could I ask for review? 😄

Sure, I'll test it tomorrow. Sorry for the late response, I thought I enabled watch for this repository but turned out I didn't 😅

File: pysnmp/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # http://www.python.org/dev/peps/pep-0396/
-__version__ = '4.4.12'
+__version__ = "5.0.10"
```
Comment (omrozowicz-splunk) [inline pysnmp/__init__.py:2]: Can you update it to `5.0.16`? That would be the released version after merge. We'll make it increase automatically soon.

**Review by omrozowicz-splunk — CHANGES_REQUESTED** [review]

**Review by omrozowicz-splunk — APPROVED** [review]

**Review by omrozowicz-splunk — CHANGES_REQUESTED** [review]
Please pull form the current main (conflict in pysnmp/__init__.py) and I'll merge you PR.

**Comment by LissaGreense** [comment]
@omrozowicz-splunk thanks for your time! :smile: I changed version 5.0.16 as you asked ^^

File: pysnmp/proto/secmod/rfc3414/service.py [hunk 2]
```diff
@@ -446,34 +446,40 @@ def __generateRequestOrResponseMsg(self, snmpEngine,
...
-            snmpEngineBoots = snmpEngineTime = 0
-            debug.logger & debug.flagSM and debug.logger(
-                '__generateRequestOrResponseMsg: assuming zero snmpEngineBoots, snmpEngineTime')
+                    '__generateRequestOrResponseMsg: assuming zero snmpEngineBoots, snmpEngineTime')
 
-        debug.logger & debug.flagSM and debug.logger(
-            '__generateRequestOrResponseMsg: use snmpEngineBoots {} snmpEngineTime {} for securityEngineID {!r}'.format(
-                snmpEngineBoots, snmpEngineTime, securityEngineID))
+            debug.logger & debug.flagSM and debug.logger(
+                '__generateRequestOrResponseMsg: use snmpEngineBoots %s snmpEngineTime %s for securityEngineID %r' % (
```
Comment (omrozowicz-splunk) [inline pysnmp/proto/secmod/rfc3414/service.py:481]: Ok, last thing is to make `pre-commit` step green. It requires running `pre-commit run --all-files` command in the root of the repository. It shows one error in this line. You can check the details here -> https://github.com/pysnmp/pysnmp/runs/6724646831?check_suite_focus=true

Comment (LissaGreense) [reply pysnmp/proto/secmod/rfc3414/service.py:481]: Ouu... thanks, I'm checking it! :smile:

Comment (LissaGreense) [reply pysnmp/proto/secmod/rfc3414/service.py]: Done ;)

**Pull request merged**